plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -619,6 +619,9 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
         GHC.Var (isErrorId -> True) `GHC.App` GHC.Type t `GHC.App` _ ->
             PIR.TyInst () <$> errorFunc <*> compileTypeNorm t
 
+        -- Remove traces according to a plugin flag
+        GHC.Var f `GHC.App` _ `GHC.App` _ `GHC.App` a | coRemoveTrace compileOpts && GHC.getOccString f == "trace" -> compileExpr a
```

Comment:
This is the whole thing. The casing on the name is pretty scary, I wonder if we could/should be more precise here?

Reply:
Yeah, definitely not. We should at least use the GHC names. 

Looking at this, I can think of a much neater way. In `Builtins` we define what the implementation of various builtins should be. That already sets things up based on the names properly. We can just define `trace` to be `\s a -> a` instead of actually trace. Then it's an optimization problem to inline that and clear things up.

Reply:
i.e. here: https://github.com/input-output-hk/plutus/blob/master/plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs#L306

Reply:
True, that was a very lazy attempt, have updated as suggested!